StrategicOperations/StrategicOperations/Framework/Utils.cs

```diff
@@ -926,6 +937,35 @@ public static float GetAvoidStrafeChanceForTeam(this ICombatant combatant)
             return finalAA;
         }
 
+        private static float GetAvoidStrafeChanceForTeamAlternate(this ICombatant combatant, string attackingUnitId)
+        {
+            var actors = combatant.Combat.GetAllLivingActors();
+            var cumAA = 0f;
+            foreach (var unit in actors)
+            {
+                if (unit.team.IsFriendly(combatant.team))
+                {
+                    var distance = (combatant.CurrentPosition - unit.CurrentPosition).magnitude;
+                    if (distance <= ModInit.modSettings.strafeAAMaxCoverDistance)
+                    {
+                        cumAA += unit.GetAAAFactor();
+                        ModInit.modLog?.Debug?.Write($"unit {unit.DisplayName} is friendly of {combatant.DisplayName} at distance {distance} which is within maximum cover distance of {ModInit.modSettings.strafeAAMaxCoverDistance}. " +
+                                                     $"Added AA factor {unit.GetAAAFactor()}; total is now {cumAA}");
+                    }
+                }
+            }
+
+            if (!ModInit.modSettings.strafeAttackerStrength.TryGetValue(attackingUnitId, out var strafeAttackerDetermination))
+            {
+                ModInit.modLog?.Debug?.Write($"No strafe attacker strength found for {attackingUnitId}, using fallback.");
+                strafeAttackerDetermination = 5f;
```

Comment:
we should make the fallback a modsetting so we can adjust it easily if we need to balance or change it down the line